lib/libzfs/libzfs_dataset.c

```diff
@@ -196,6 +196,16 @@ zfs_validate_name(libzfs_handle_t *hdl, const char *path, int type,
 				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
 				    "reserved disk name"));
 				break;
+			
+			case NAME_ERR_SELF_MOUNT:
+				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+				    "self reference is found"));
```

Comment:
I think this error message can be clearer. Perhaps "filesystem name references self"?

Reply:
How about NAME_ERR_SELF_REFERENCE and NAME_ERR_PARENT_REFERENCE

Reply:
I apologize for the confusion. I like the enum names. My comments referred to the comments or error message text.

Reply:
No problem, I feel they were not clear NAME_ERR_SELF_REFERENCE and NAME_ERR_PARENT_REFERENCE makes more sense.